src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -374,35 +396,60 @@ private static CastResult TryGet(nuint source, nuint target)
         [DebuggerStepThrough]
         private static object? ChkCastInterface(void* toTypeHnd, object? obj)
         {
+            const int unrollSize = 4;
+
             if (obj != null)
             {
                 MethodTable* mt = RuntimeHelpers.GetMethodTable(obj);
-                nuint interfaceCount = mt->InterfaceCount;
+                nint interfaceCount = mt->InterfaceCount;
                 if (interfaceCount == 0)
                 {
                     goto slowPath;
                 }
 
                 MethodTable** interfaceMap = mt->InterfaceMap;
-                for (nuint i = 0; ; i += 4)
+                if (interfaceCount < unrollSize)
                 {
-                    if (interfaceMap[i + 0] == toTypeHnd)
-                        goto done;
-                    if (--interfaceCount == 0)
-                        goto slowPath;
-                    if (interfaceMap[i + 1] == toTypeHnd)
-                        goto done;
-                    if (--interfaceCount == 0)
-                        goto slowPath;
-                    if (interfaceMap[i + 2] == toTypeHnd)
-                        goto done;
-                    if (--interfaceCount == 0)
-                        goto slowPath;
-                    if (interfaceMap[i + 3] == toTypeHnd)
+                    // If not enough for unrolled, jmp straight to small loop
+                    // as we already know there is one or more interfaces so don't need to check again.
+                    goto few;
```

Comment:
If interface table is allocated as 4 elements aligned and zero-filled, we could remove the "few" case entirely. At least in Chk case which is expected to succeed.
That would come with size increase though, so does not seem worth it.